libdnf/sack/query.cpp

```diff
@@ -1008,10 +1008,17 @@ Query::Impl::initResult()
             result->set(solvid);
         dnf_sack_set_pkg_solvables(sack, result->getMap(), pool->nsolvables);
     }
-    if (!(flags & HY_IGNORE_EXCLUDES)) {
+    if (flags == HY_APPLY_EXCLUDES) {
         dnf_sack_recompute_considered(sack);
         if (pool->considered)
             map_and(result->getMap(), pool->considered);
+    } else {
+        Map * considered = nullptr;
+        dnf_sack_recompute_considered_map(sack, &considered, static_cast<hy_exclude_flags>(flags));
+        if (considered) {
+            map_and(result->getMap(), considered);
+            free_map_fully(considered);
+        }
```

Comment:
Will this work well? It seems you don't want to store partial excludes into `pool->considered`, but can't it potentially cause inconsistencies later?

Reply:
I will think about cassing